dotCMS/src/main/java/com/dotcms/rendering/velocity/services/ContentletLoader.java

```diff
@@ -0,0 +1,851 @@
+
+        if (PageMode.EDIT == mode) {
+            sb.append("#set( $EDIT_CONTENT_PERMISSION=$EDIT_CONTENT_PERMISSION")
+                .append(identifier.getId())
+                .append(" )");
+        }
+
+        sb.append("#set( $CONTENT_INODE=\"")
+            .append(content.getInode())
+            .append("\" )");
+        sb.append("#set( $IDENTIFIER_INODE=\"")
+            .append(identifier.getId())
+            .append("\" )");
+
+        sb.append("#set( $ContentInode=\"")
+            .append(content.getInode())
+            .append("\" )");
+        sb.append("#set( $ContentIdentifier=\"")
+            .append(identifier.getId())
+            .append("\" )");
+        sb.append("#set( $ContentletTitle=\"")
+            .append(UtilMethods.espaceForVelocity(conAPI.getName(content, APILocator.getUserAPI()
+                .getSystemUser(), true)))
+            .append("\" )");
+        sb.append("#set( $ContentletStructure=\"")
+            .append(content.getContentTypeId())
+            .append("\" )");
+        sb.append("#set( $ContentletContentType=\"")
+            .append(content.getContentTypeId())
+            .append("\" )");
+        if (type.baseType() == BaseContentType.WIDGET) {
+            sb.append("#set( $isWidget= \"")
+                .append(true)
+                .append("\")");
+            if (type.name()
+                .equals(FormAPI.FORM_WIDGET_STRUCTURE_NAME_FIELD_NAME)) {
+                sb.append("#set($isFormWidget= \"")
+                    .append(true)
+                    .append("\")");
+            } else {
+                sb.append("#set($isFormWidget= \"")
+                    .append(false)
+                    .append("\")");
+            }
+        } else {
+            sb.append("#set($isWidget= \"")
+                .append(false)
+                .append("\")");
+        }
+
+        // close contentlet div
+        sb.append("#if($API_EDIT_MODE)")
+            .append("</div>")
+            .append("#end");
+
+        return writeOutVelocity(filePath, sb.toString());
+    }
+
+    @SuppressWarnings("unchecked")
+    public List<Contentlet> sortContentlets(List<Contentlet> contentletList, String sortBy) {
+
+        Logger.debug(this.getClass(), "I'm ordering by " + sortBy);
+
+        if (sortBy.equals("tree_order")) {
+            Collections.sort(contentletList, new WebAssetSortOrderComparator());
+        } else {
+            Collections.sort(contentletList, new ContentComparator("asc"));
+        }
+
+        return contentletList;
+
+    }
+
+    /**
+     * Will remove all contentlet files within a structure for both live and working. Uses the
+     * system user.
+     * 
+     * @param contentlets
+     * @throws DotSecurityException
+     * @throws DotDataException
+     */
+    public void invalidate(Structure structure) throws DotDataException, DotSecurityException {
+        ContentletAPI conAPI = APILocator.getContentletAPI();
+        int limit = 500;
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Assign this magic number 500 to a well-named constant, and use the constant instead. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS109)